ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -148,17 +148,20 @@ fetchDecisions
           lastStarvationTime <- case chainSelStarvation of
             ChainSelStarvationEndedAt time -> pure time
             ChainSelStarvationOngoing -> getMonotonicTime
-          if lastStarvationTime < addTime bulkSyncGracePeriod peersOrderStart
-            then pure peersOrder
-            else do
-              let peersOrder' =
-                    PeersOrder
-                      { peersOrderCurrent = Nothing,
-                        peersOrderOthers = msnoc peersOrderOthers peersOrderCurrent,
-                        peersOrderStart
-                      }
-              traverse_ demoteCSJDynamoAndIgnoreInflightBlocks peersOrderCurrent
-              pure peersOrder'
+          case peersOrderCurrent of
+            Just peersOrderCurrent_
+              | peerHasBlocksInFlight peersOrderCurrent_
+                  && lastStarvationTime >= addTime bulkSyncGracePeriod peersOrderStart ->
```

Comment:
This condition is being reversed! But it looks to me like the new version is the correct one.

Reply:
Compared to previous commits, you mean? I actually forgot to flip the sign, initially, and it was caught by the tests.

Reply:
I do think it is the right direction.